### PR TITLE
Build OpenLayers js dependency

### PR DESCRIPTION
**Pull request by ahocevar**
See openlayers/openlayers#11826 for the problem.

This pull request adds a simple webpack config to create an optimized OpenLayers build with just the modules that are needed. These modules are added to `./js/src/ol.js`. And if any additional functionality of OpenLayers is required in the future, this is where the additional modules have to be added.

As a result, OpenLayers no longer needs to be manually updated - it is now updated as part of the `./scripts/sync-js-vendor-files.sh` task.

### CODE REVIEW

**Comment by ibennetch** [comment]
This seems like a great improvement and big help to the build/maintenance toolchain.

It looks good to me, but since @mauriciofauth is more familiar with this workflow, let's see if he has any comment.

**Review by williamdes — APPROVED** [review]
I really like this improvement!
I would prefer if it is possible to only have web pack installed on the fly in a temporary folder to build the files as we do not use it anywhere else for now
Not sure about this but I still write it down in case somebody else thinks it makes sense

**Comment by ahocevar** [comment]
I added another commit so `webpack` is no longer required as dependency. Instead, `npx` is used to temporarily install `webpack-cli`. This requires the developer to confirm "yes" when running the `sync-js-vendor-files.sh` script.

**Review by williamdes — APPROVED** [review]
Brilliant

**Comment by williamdes** [comment]
If nobody objects I would send this into Debian
We **have** to have reproducible js bundles at Debian

**Comment by ahocevar** [comment]
I added another commit to fix the `js-lint` failure.

File: scripts/sync-js-vendor-files.sh [hunk 1]
```diff
@@ -1,4 +1,5 @@
-#!/bin/bash
+#!/bin/bash -eu
```
Comment (williamdes) [inline scripts/sync-js-vendor-files.sh:1]: Added this even if this is un-related but it is a security fix

File: scripts/sync-js-vendor-files.sh [hunk 2]
```diff
@@ -13,9 +14,7 @@ cd ${ROOT_DIR}
 
 # Remove each '-not -path' when a new package can be used from npm
 echo 'Delete vendor files we can replace from source dists'
-find ./js/vendor/ \
-    -not -path './js/vendor/openlayers/*' \
-    -type f -delete -print
```
Comment (williamdes) [inline scripts/sync-js-vendor-files.sh:18]: This one was missed, I added it. Now we are sure 100% of the files are re-built, this showed 2 files needed to be deleted.

File: scripts/sync-js-vendor-files.sh [hunk 3]
```diff
@@ -74,8 +73,20 @@ echo 'Updating jquery-debounce'
...
+echo 'Updating OpenLayers'
+cp ./node_modules/ol/ol.css ./js/vendor/openlayers/theme/ol.css
+npx webpack-cli --config ./js/config/ol/webpack.config.js
+echo "/*!
+  * OpenLayers v$(yarn -s info ol version) (https://openlayers.org/)
+  * Copyright 2005-present, OpenLayers Contributors All rights reserved.
+  * Licensed under BSD 2-Clause License (https://github.com/openlayers/openlayers/blob/main/LICENSE.md)
+  *
+  * @license $(yarn -s info ol license)
+  */
```
Comment (williamdes) [inline scripts/sync-js-vendor-files.sh:85]: @ahocevar can you confirm the license info looks good ?

Comment (ahocevar) [reply scripts/sync-js-vendor-files.sh:85]: Yes, this looks good.

File: scripts/sync-js-vendor-files.sh (hunk 2 above)
Comment (ahocevar) [reply scripts/sync-js-vendor-files.sh:18]: Yes, that makes sense. Thanks! The js files are generated by this script, and the css files are copied. So there is no need any more to exclude `./js/vendor/openlayers/*` from the deletion process.

**Comment by williamdes** [comment]
About what I said for Debian, [it builds ](https://salsa.debian.org/phpmyadmin-team/phpmyadmin/-/jobs/1701781#L3295) using the diff from this PR and [a bunch of Debian changes](https://salsa.debian.org/phpmyadmin-team/phpmyadmin/-/compare/cdd0ebfaceb368b1c9494dde4c0879f3ff85b7d7...940d451dc0fe5d8d7ef45c2e0597ab3b1495b263
)

To see the Debian commits: https://salsa.debian.org/phpmyadmin-team/phpmyadmin/-/compare/cdd0ebfaceb368b1c9494dde4c0879f3ff85b7d7...940d451dc0fe5d8d7ef45c2e0597ab3b1495b263

```
# build openlayers
webpack --config ./js/config/ol/webpack.config.js
Hash: c2c3e74ac3f667f74df2
Version: webpack 4.30.0
Time: 5972ms
Built at: 13/06/2021 23:21:42
            Asset      Size  Chunks             Chunk Names
    OpenLayers.js   294 KiB       0  [emitted]  main
OpenLayers.js.map  1.75 MiB       0  [emitted]  main
Entrypoint main = OpenLayers.js OpenLayers.js.map
 [0] ./debian/missing-sources/ol/control/FullScreen.js 9.64 KiB [built]
 [1] ./debian/missing-sources/ol/control/OverviewMap.js 21 KiB [built]
 [2] ./debian/missing-sources/ol/control/ScaleLine.js 15.8 KiB [built]
 [3] ./debian/missing-sources/ol/control/ZoomSlider.js 12.1 KiB [built]
 [4] ./debian/missing-sources/ol/control/ZoomToExtent.js 3.6 KiB [built]
[13] ./debian/missing-sources/ol/style.js 558 bytes [built]
[14] ./js/src/ol.mjs + 189 modules 1.29 MiB {0} [built]
     | ./js/src/ol.mjs 1.03 KiB [built]
     | ./debian/missing-sources/ol/extent.js 23 KiB [built]
     | ./debian/missing-sources/ol/coordinate.js 12.5 KiB [built]
     | ./debian/missing-sources/ol/proj.js 25.1 KiB [built]
     | ./debian/missing-sources/ol/control.js 2.65 KiB [built]
     | ./debian/missing-sources/ol/Disposable.js 773 bytes [built]
     | ./debian/missing-sources/ol/Observable.js 4.95 KiB [built]
     | ./debian/missing-sources/ol/util.js 857 bytes [built]
     | ./debian/missing-sources/ol/Object.js 7.91 KiB [built]
     | ./debian/missing-sources/ol/control/Control.js 5.6 KiB [built]
     | ./debian/missing-sources/ol/AssertionError.js 2.05 KiB [built]
     | ./debian/missing-sources/ol/asserts.js 346 bytes [built]
     | ./debian/missing-sources/ol/math.js 5.7 KiB [built]
     | ./debian/missing-sources/ol/layer/Layer.js 12.2 KiB [built]
     | ./debian/missing-sources/ol/control/Attribution.js 11.2 KiB [built]
     |     + 175 hidden modules
[15] ./debian/missing-sources/ol/geom.js 751 bytes [built]
[16] ./debian/missing-sources/ol/index.js 1.52 KiB [built]
[17] ./debian/missing-sources/ol/layer.js 707 bytes [built]
[18] ./debian/missing-sources/ol/source.js 1.62 KiB [built]
[19] ./debian/missing-sources/ol/Geolocation.js 11.6 KiB [built]
[20] ./debian/missing-sources/ol/ImageCanvas.js 3.22 KiB [built]
[21] ./debian/missing-sources/ol/Overlay.js 19.3 KiB [built]
[23] ./debian/missing-sources/ol/VectorRenderTile.js 5.67 KiB [built]
    + 63 hidden modules
```

**Pull request merged**